ios/RPCModule.swift

```diff
@@ -11,294 +11,346 @@ import React
 @objc(RPCModule)
 class RPCModule: NSObject {
   
+  private let walletFileName = "wallet.dat.txt"
+  private let walletBackupFileName = "wallet.backup.dat.txt"
+  private let backgroundFileName = "background.json"
+  private let errorPrefix = "error"
+  
   @objc
   static func requiresMainQueueSetup() -> Bool {
       return true
   }
   
+  enum FileError: Error {
+    case documentsDirectoryNotFoundError(String)
+    case readWalletUtf8StringError(String)
+    case readWalletDecodedDataError(String)
+    case saveFileDecodingError(String)
+    case writeFileError(String)
+    case deleteFileError(String)
+  }
+  
+  func getDocumentsDirectory() throws -> String {
+    let paths = NSSearchPathForDirectoriesInDomains(.documentDirectory, .userDomainMask, true)
+    guard let pathsFirst = paths.first else {
+      throw FileError.documentsDirectoryNotFoundError("Documents directory could not be located.")
+    }
+    return pathsFirst
+  }
+  
+  func getFileName(_ file: String) throws -> String {
+    let documentsDirectory = try getDocumentsDirectory()
+    let fileName = "\(documentsDirectory)/\(file)"
+    return fileName
+  }
+  
+  func fileExists(_ fileName: String) -> String {
+    let fileExists = FileManager.default.fileExists(atPath: fileName)
+    if fileExists {
+      return "true"
+    } else {
+      return "false"
+    }
+  }
+  
   @objc(walletExists:reject:)
   func walletExists(_ resolve: @escaping RCTPromiseResolveBlock, reject: @escaping RCTPromiseRejectBlock) {
-      let paths = NSSearchPathForDirectoriesInDomains(.documentDirectory, .userDomainMask, true)
-      if let documentsDirectory = paths.first {
-        let fileName = "\(documentsDirectory)/wallet.dat.txt"
-        let fileExists = FileManager.default.fileExists(atPath: fileName)
-        if fileExists {
-            resolve("true")
-        } else {
-            resolve("false")
-        }
-      } else {
-        resolve("false")
-      }
+    do {
+      let fileName = try getFileName(walletFileName)
+      resolve(fileExists(fileName))
+    } catch {
+      NSLog("wallet exists error: \(error.localizedDescription)")
+      resolve("false")
+    }
   }
 
   @objc(walletBackupExists:reject:)
   func walletBackupExists(_ resolve: @escaping RCTPromiseResolveBlock, reject: @escaping RCTPromiseRejectBlock) {
-      let paths = NSSearchPathForDirectoriesInDomains(.documentDirectory, .userDomainMask, true)
-      if let documentsDirectory = paths.first {
-        let fileName = "\(documentsDirectory)/wallet.backup.dat.txt"
-        let fileExists = FileManager.default.fileExists(atPath: fileName)
-        if fileExists {
-            resolve("true")
-        } else {
-            resolve("false")
-        }
-      } else {
-        resolve("false")
-      }
+    do {
+      let fileName = try getFileName(walletBackupFileName)
+      resolve(fileExists(fileName))
```

Comment:
Are there two different ways to `resolve("false")` here?

Reply:
Yes, `fileExists` can return `false` or `true` because the wallet file can exists or not, and if `try getFileName` throw an error, then I return `false` as well.